src/dis/SignalPdu.js

```diff
@@ -8,116 +8,110 @@
-  dis.SignalPdu.prototype.encodeToBinary = function(outputStream)
-  {
-       outputStream.writeUByte(this.protocolVersion);
-       outputStream.writeUByte(this.exerciseID);
-       outputStream.writeUByte(this.pduType);
-       outputStream.writeUByte(this.protocolFamily);
-       outputStream.writeUInt(this.timestamp);
-       outputStream.writeUShort(this.pduLength);
-       outputStream.writeShort(this.padding);
-       this.entityId.encodeToBinary(outputStream);
-       outputStream.writeUShort(this.radioId);
-       outputStream.writeUShort(this.encodingScheme);
-       outputStream.writeUShort(this.tdlType);
-       outputStream.writeUInt(this.sampleRate);
-       outputStream.writeUShort(this.dataLength);
-       outputStream.writeUShort(this.samples);
-       for(var idx = 0; idx < this.samples; idx++)
-       {
-        this.data[idx].encodeToBinary(outputStream);
-       }
-
-  };
+	exports = {};
+
+
+dis.SignalPdu = function () {
+	/** The version of the protocol. 5=DIS-1995, 6=DIS-1998. */
+	this.protocolVersion = 6;
+
+	/** Exercise ID */
+	this.exerciseID = 0;
+
+	/** Type of pdu, unique for each PDU class */
+	this.pduType = 26;
+
+	/** value that refers to the protocol family, eg SimulationManagement, et */
+	this.protocolFamily = 4;
+
+	/** Timestamp value */
+	this.timestamp = 0;
+
+	/** Length, in bytes, of the PDU. Changed name from length to avoid use of Hibernate QL reserved word */
+	this.pduLength = 0;
+
+	/** zero-filled array of padding */
+	this.padding = 0;
+
+	/** ID of the entity that is the source of the communication, ie contains the radio */
+	this.entityId = new dis.EntityID();
+
+	/** particular radio within an entity */
+	this.radioId = 0;
+
+	/** encoding scheme used, and enumeration */
+	this.encodingScheme = 0;
+
+	/** tdl type */
+	this.tdlType = 0;
+
+	/** sample rate */
+	this.sampleRate = 0;
+
+	/** length of data, in bits */
+	this.dataLength = 0;
+
+	/** number of samples. If the PDU contains encoded audio, this should be zero. */
+	this.samples = 0;
+
+	/** list of eight bit values. Must be padded to fall on a 32 bit boundary. */
+	this.data = new Array();
+
+	dis.SignalPdu.prototype.initFromBinary = function (inputStream) {
+		this.protocolVersion = inputStream.readUByte();
+		this.exerciseID = inputStream.readUByte();
+		this.pduType = inputStream.readUByte();
+		this.protocolFamily = inputStream.readUByte();
+		this.timestamp = inputStream.readUInt();
+		this.pduLength = inputStream.readUShort();
+		this.padding = inputStream.readShort();
+		this.entityId.initFromBinary(inputStream);
+		this.radioId = inputStream.readUShort();
+		this.encodingScheme = inputStream.readUShort();
+		this.tdlType = inputStream.readUShort();
+		this.sampleRate = inputStream.readUInt();
+		this.dataLength = inputStream.readUShort();
+		this.samples = inputStream.readUShort();
+		try {
+			for (var idx = 0; idx < (this.dataLength / 8); idx++) {
+				var anX = new dis.Chunk(1);
+				anX.initFromBinaryDIS(inputStream);
```

Comment:
@Klumze this change to use `initFromBinaryDIS`, remind me what this one was about?

Reply:
Lol I think this is just cause I had beautified the file to see things better, that change can be discarded if needed

Reply:
Ok yes lets discard that change if it's not required.